test/tests.luau

```diff
@@ -1012,6 +1012,28 @@ TEST("world:target", function()
         CHECK(world:target(e, C, 1) == nil)
     end
 
+    do CASE "infer index when unspecified"
+        local world = world_new()
+        local A = world:component()
+        local B = world:component()
+        local C = world:component()
+        local D = world:component()
+        local e = world:entity()
+
+        world:add(e, pair(A, B))
+        world:add(e, pair(A, C))
+        world:add(e, pair(B, C))
+        world:add(e, pair(B, D))
+        world:add(e, pair(C, D))
+
+        CHECK(world:target(e, A) == B)
```

Comment:
should be an extra path that checks `world:target(e, X) == world:target(e, X, 0)`

Reply:
[done](https://github.com/Ukendio/jecs/pull/131/commits/4de22737a915b71597317019ed7827ea8da6767d)